Extension/src/LanguageServer/configurations.ts

```diff
@@ -2029,7 +2029,17 @@ export class CppProperties {
             // and extend that pattern to the next quote before and next quote after it.
             const pattern: RegExp = new RegExp(`"[^"]*?(?<="|;)${escapedPath}(?="|;).*?"`, "g");
             const configMatches: string[] | null = curText.match(pattern);
-            if (configMatches) {
+
+            let globPath: boolean = false;
+            const asteriskPosition = curPath.indexOf("*");
+            if (asteriskPosition !== -1 && asteriskPosition !== curPath.length - 1 && asteriskPosition !== curPath.length - 2) {
```

Comment:
The case "a" doesn't report a squiggle (corner case). You can fix this via moving the check `asteriskPosition !== -1` into an outer check and put the other checks inside that, i.e. the problem is `asteriskPosition === curPath.length - 2` is being checked when `asteriskPosition !== -1`.